src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -126,6 +166,13 @@ class SpfxProjectUpgradeCommand extends Command {
       try {
         const rules: Rule[] = require(`./project-upgrade/upgrade-${v}`);
         rules.forEach(r => {
+          if (r instanceof FN012017_TSC_extends) {
```

Comment:
Unfortunately, this approach won't work. If there is no difference related to TypeScript let's say between v1.8.0 and v1.8.1, neither rule will be returned in the list of rules and you wouldn't be able to update your project's TypeScript version. The collection of rules returned for upgrading the project to the specific version only returns the delta rules rather than all rules.